src/ast.rs

```diff
@@ -181,6 +296,12 @@ impl From<Scan> for Statement {
     }
 }
 
+impl DisplayWithContext for Scan {
+    fn fmt(&self, f: &mut fmt::Formatter, ctx: &Context) -> fmt::Result {
+        write!(f, "scan {} {{ ... }}", self.value.display_with(ctx))
```

Comment:
:smile_cat:

Reply:
This is not an omission :). My thinking is that we format for error messages, and not for pretty-printing a parsible AST. The scan statements, and arms, would be pretty large. So they'll become very long lines (which still wouldn't be valid syntax), or multi-line (with all formatting problems of multi-line errors), and the extra information will not improve the debugging experience much. I am making sure that the arm and statement where the error originated is also included in the context. And, if we have line numbers in the error as well, it will be easy to find in the source, even without the details of the arms.

An example of a trace that originated in a scan arm:
```
Caused by:
    0: Executing scan "a/b/c.py" { ... }
    1: Matching a/b/c.py with arm "([^/]+/)+([^/]*).py$" { ... }
    2: Executing set x = $2
    3: Undefined variable x
```